### PR TITLE
feat: extend customizability of the deployment

### PR DESCRIPTION
**Pull request by maxgio92**

### CODE REVIEW

File: values.yml [hunk 1]
```diff
@@ -1,11 +1,8 @@
 #@data/values
 ---
-#@overlay/match missing_ok=True
```
Comment (prometherion) [inline values.yml:3]: Why removing the missing_ok tokens here?

**Review by prometherion — COMMENTED** [review]
How did you test the YTT rendering?

File: values.yml (hunk 1 above)
Comment (maxgio92) [reply values.yml:3]: Because they are defined in the `values-schema` in the `schema.yml` file, and we put values only in `values`

**Comment by maxgio92** [comment]
> How did you test the YTT rendering?

I tested it with a pipeline following this order:
1. https://github.com/clastix/src-to-oci-build
2. https://github.com/clastix/src-to-oci-deploy
3. inline data values

For example:
```sh
git clone https://github.com/clastix/src-to-oci-build
git clone https://github.com/clastix/src-to-oci-deploy

ytt -f src-to-oci-build -f src-to-oci-deploy # failing
ytt -f src-to-oci-build -f src-to-oci-deploy -v=app_name=myapp # also failing
ytt -f src-to-oci-build -f src-to-oci-deploy -v=programming_language_runtime=go # also failing
ytt -f src-to-oci-build -f src-to-oci-deploy -v=app_name=myapp -v=programming_language_runtime=go # passing
```

WDYT?

**Comment by maxgio92** [comment]
Edit: this order doesn't seem to be the same used by the `kapp-controller` if specifying an App configured like that:

```
apiVersion: kappctrl.k14s.io/v1alpha1
kind: App
metadata:
  name: go-simpleapp
  namespace: default
spec:
  serviceAccountName: kapp-app-simpleapp
  fetch:
  - git:
      url: https://github.com/clastix/src-to-oci-go-simpleapp
      ref: origin/main
  - git:
      url: https://github.com/clastix/src-to-oci-build
      ref: origin/feat/customize
  - git:
      url: https://github.com/clastix/src-to-oci-deploy
      ref: origin/feat/customize
  template:
  - ytt:
      paths:
      - "./1"
      - "./2"
      inline:
        paths:
          values.yml: |-
            #@data/values
            ---
            app_name: myapp
            programming_language_runtime: go
            push_images_repo: quay.io/maxgio92/go-simpleapp
            replicas: 1
  - kbld: {}
  deploy:
  - kapp: {}
```

Putting the PR in WIP state.

**Review by prometherion — APPROVED** [review]

**Pull request merged**